pkg/virt-controller/watch/network.go

```diff
@@ -24,23 +24,29 @@ import (
 	"kubevirt.io/kubevirt/pkg/network/vmispec"
 )
 
-func calculateDynamicInterfaces(vmi *v1.VirtualMachineInstance) ([]v1.Interface, []v1.Network, bool) {
-	vmiSpecIfaces := vmispec.FilterInterfacesSpec(vmi.Spec.Domain.Devices.Interfaces, func(iface v1.Interface) bool {
+func calculateInterfacesAndNetworksForMultusAnnotation(vmi *v1.VirtualMachineInstance) ([]v1.Interface, []v1.Network, bool) {
+	nonAbsentSpecIfaces := vmispec.FilterInterfacesSpec(vmi.Spec.Domain.Devices.Interfaces, func(iface v1.Interface) bool {
 		return iface.State != v1.InterfaceStateAbsent
 	})
-	ifacesToHotUnplugExist := len(vmi.Spec.Domain.Devices.Interfaces) > len(vmiSpecIfaces)
+	nonAbsentSpecNetworks := vmispec.FilterNetworksByInterfaces(vmi.Spec.Networks, nonAbsentSpecIfaces)
 
-	vmiSpecNets := vmi.Spec.Networks
-	if ifacesToHotUnplugExist {
-		vmiSpecNets = vmispec.FilterNetworksByInterfaces(vmi.Spec.Networks, vmiSpecIfaces)
-	}
-	ifacesToHotplugExist := len(vmispec.NetworksToHotplug(vmiSpecNets, vmi.Status.Interfaces)) > 0
+	networksToHotplug := vmispec.NetworksToHotplug(nonAbsentSpecNetworks, vmi.Status.Interfaces)
+	indexedNetworksToHotplug := vmispec.IndexNetworkSpecByName(networksToHotplug)
+	nonAbsentIfacesWithotSriovHotplug := vmispec.FilterInterfacesSpec(nonAbsentSpecIfaces, func(iface v1.Interface) bool {
+		_, needsToBeHotplugged := indexedNetworksToHotplug[iface.Name]
+		return !needsToBeHotplugged || iface.InterfaceBindingMethod.SRIOV == nil
+	})
+	nonAbsentNetworksWithoutSriovHotplug := vmispec.FilterNetworksByInterfaces(nonAbsentSpecNetworks, nonAbsentIfacesWithotSriovHotplug)
 
+	// SRIOV interfaces cannot be hotplugged in place since the device plugin cannot be invoked after the pod startup
+	numOfSriovIfacesToHotplug := len(nonAbsentSpecIfaces) - len(nonAbsentIfacesWithotSriovHotplug)
```

Comment:
I am having hard time to understand the how the filtering of  SR-IOV interface that should be hotplug is performed.

I am having hard time to understand this function strategy, spesificly how SR-IOV interfaces to hotplug are filtered.
Could you please explain this flow briefly?